packages/gatsby-cli/src/init-starter.js

```diff
@@ -67,8 +71,30 @@ const install = async rootPath => {
   process.chdir(rootPath)
 
   try {
-    let cmd = shouldUseYarn() ? spawn(`yarnpkg`) : spawn(`npm install`)
-    await cmd
+    const npmCmd = `npm install`
+    let response = npmCmd
+    if (shouldUseYarn()) {
+      const promptsAnswer = await prompts([
+        {
+          type: `select`,
+          name: `package_manager`,
+          message: `Which package manager would you like to use ?`,
+          choices: [
+            { title: `yarn`, value: `yarnpkg` },
+            { title: `npm`, value: npmCmd },
+          ],
+          initial: 0,
+        },
+      ])
+      response = promptsAnswer.package_manager
+    }
+    conf.set(`package_manager`, response)
+    if (response.includes(`yarn`)) {
+      fs.unlinkSync(`package-lock.json`)
```

Comment:
async removal of files has better support on windows as it retries if it fails on the first try
```suggestion
      await fs.remove(`package-lock.json`)
```

https://github.com/jprichardson/node-fs-extra/blob/master/lib/remove/rimraf.js#L45-L51